### ISSUE TITLE
issue with fix_inputs and default equivalencies

### ISSUE THREAD

**Issue by nden** [astropy/astropy#10319]
`Model.input_units_equivalencies` crashes when the model is the result of `fix_inputs` operation and the constituent models have no equivalencies set.

```
p = Polynomial1D(1, c0=1, c1=2)
m = p & p
mf = fix_inputs(m, {'x0': 1})
mf.input_units_equivalencies                                                 
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-59-92566a4c98dd> in <module>
----> 1 mf.input_units_equivalencies

~/dev/astropy/astropy/modeling/core.py in input_units_equivalencies(self)
   3128         inputs_map = self.inputs_map()
   3129         return {key: inputs_map[key][0].input_units_equivalencies[orig_key]
-> 3130                 for key, (mod, orig_key) in inputs_map.items()
   3131                 if inputs_map[key][0].input_units_equivalencies is not None}
   3132 

~/dev/astropy/astropy/modeling/core.py in <dictcomp>(.0)
   3129         return {key: inputs_map[key][0].input_units_equivalencies[orig_key]
   3130                 for key, (mod, orig_key) in inputs_map.items()
-> 3131                 if inputs_map[key][0].input_units_equivalencies is not None}
   3132 
   3133     @property

KeyError: 'x1'

```
It looks like `fix_inputs` changes the default value of `input_units_equivalencies` of the constituent models from `None` to `{}`. This results in a crash when trying to access the property.
From the above example, accessing the property on the child models gives different results.
```
m.inputs_map()['x1'][0].input_units_equivalencies
None

mf.inputs_map()['x1'][0].input_units_equivalencies
{}
```

**Comment by Iamsoto** [comment]
Hi,

I'm new here. Just trying to help out. 

Is it possible to change the return on **input_units_equivalencies** from :
`return {key: inputs_map[key][0].input_units_equivalencies[orig_key]
                for key, (mod, orig_key) in inputs_map.items()
                if inputs_map[key][0].input_units_equivalencies is not None}` 

to 

`return {key: inputs_map[key][0].input_units_equivalencies[orig_key]
                for key, (mod, orig_key) in inputs_map.items()
                if inputs_map[key][0].input_units_equivalencies is not None and orig_key in inputs_map[key][0].input_units_equivalencies}`

**Comment by Iamsoto** [comment]
Hi! 

I'm aware the fix I mentioned above addresses the first part of your question, but not the latter.
That is, why: 

```
m.inputs_map()['x1'][0].input_units_equivalencies
None

mf.inputs_map()['x1'][0].input_units_equivalencies
{}
```

So I did some digging and this is what I've found out. 
`CompoundModel` has the property function` input_units_equivalencies` but the Base `Model` class simply has the attribute `input_units_equivalencies = None` . 

Why does this matter? Let's look at the recursive logic for 
`mf.inputs_map()['x1'][0].input_units_equivalencies`. Bare with me here.
. 

`Polynomial1D.input_unit_equivalencies`  (returns None)--> `CompoundModel.input_unit_equivalencies` (returns {}) --> output

Where `CompoundModel.input_unit_equivalencies` returns {} due to python's list comprehension.

And if we add one more layer of recursion, as we do when we call `fix_input( ... ).input_unit_equivalencies` we get a key error, which is where the first part of your error comes in.

So TL;DR: 
If  we don't want inconsistent default values, I BELIEVE we can change 
`input_unit_equivalencies = None in astropy/core.py Model to`
to:
`input_unit_equivalencies = {} `
As well as implement the fix I mentioned earlier to avoid any key error bugs.
But I'd like someone else's input before I change default values.

I hope this all makes sense and will be helpful.
